include/networkit/auxiliary/HashUtils.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef NETWORKIT_AUXILIARY_HASH_UTILS_HPP_
+#define NETWORKIT_AUXILIARY_HASH_UTILS_HPP_
+// networkit-format
+
+#include <utility>
+
+namespace Aux {
+
+template <typename T>
+void hashCombine(std::size_t &seed, T const &v) {
+    // https://www.boost.org/doc/libs/1_75_0/doc/html/hash/reference.html#boost.hash_combine
+    seed ^= std::hash<T>()(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+}
```

Comment:
This might need to be changed or include the [Boost Software License](https://www.boost.org/users/license.html).

Reply:
I'd rather avoid to include another license. It simply complicates matters and I wonder if there is no simpler other way.

Reply:
A simple combination of two hash values which is better than just an xor would be:
```
constexpr auto halfWidth = std::numeric_limits<size_t>::digits / 2;
auto a = std::hash<A>()(pair.first);
auto b = std::hash<B>()(pair.second);
return a ^ (b << halfWidth);
```

Reply:
I think it is enough to add something like `The Boost license applies to this function.` Boost's license is non-viral and compatible with the MIT license.

Reply:
If that's the case, then fine.

Reply:
Thank you for the input. I added a license notice to the function.